docs: added some clarification on serverless functions getting executed in a non-serverless environment (#9742)
redwoodjs/redwood: docs/docs/serverless-functions.md

description: Create, develop, and run serverless functions
---

# Serverless Functions (API Endpoints)

<!-- `redwood.toml`&mdash;`api/src/functions` by default.  -->

:::info

> You can think of serverless functions as API Endpoints, and in the future we will update the terminology used.
> 
> Originally, Redwood apps were intended to be deployed as serverless functions to AWS Lambda. Whenever a Redwood app is deployed to a 'serverful' environment such as Render or Heroku, a Fastify server gets started automatically, which automatically registers the functions in `api/src/functions` onto the server. Request adapters are also automatically configured to handle the translation between Fastify's request/reply objects to the functions' AWS Lambda signature.

Reviewer: I know you were just copying from our previous docs but now that I see it again, let's remove the mention of Heroku since we don't have automatic support for deploying to it:

```suggestion
You can think of serverless functions as API Endpoints, and in the future we'll update the terminology used.

Originally, Redwood apps were intended to be deployed as serverless functions to AWS Lambda. Whenever a Redwood app is deployed to a "serverful" environment such as Fly or Render, a Fastify server is started and your Redwood app's functions in `api/src/functions` are automatically registered onto the server. Request adapters are also automatically configured to handle the translation between Fastify's request and reply objects to the functions' AWS Lambda signature.
```
